src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -16,6 +16,38 @@ public Rectangle(Integer width, Integer length, String color) {
         this.color = color;
     }
 
+    @Override
+    public int hashCode() {
+        if (width == null || length == null || color == null) {
+            return 0;
+        }
+        int res = 17;
+        res = 31 * res + width.hashCode();
+        res = 31 * res + length.hashCode();
+        res = 31 * res + color.hashCode();
+        return res;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == this) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+        if (obj.getClass().equals(Rectangle.class)) {
+            Rectangle rectangle = (Rectangle) obj;
+            return width == null && rectangle.width == null
```

Comment:
what if only width == null, but rectangle.width don't. What will happen when it will check the equals part?